pkg/apis/core/types.go

```diff
@@ -3211,6 +3211,13 @@ type ServiceSpec struct {
 	// of peer discovery.
 	// +optional
 	PublishNotReadyAddresses bool
+
+	// topologyKeys is a preference-order list of topology keys.  If backends exist for
+	// index [0], they will always be chosen; only if no backends exist for index [0] will backends for index [1] be considered.
```

Comment:
nit: I assume this is an unbounded priority list in which case it would be clear if you explicitly state, `index [1+] be considered in the order of indices`.